src/mpc-hc/MainFrm.cpp

```diff
@@ -2340,6 +2353,11 @@ void CMainFrame::GraphEventComplete()
         bBreak = !!(s.nCLSwitches & CLSW_AFTERPLAYBACK_MASK);
     }
 
+    if (!m_pSubStreams.IsEmpty() && s.fEnableSubtitles && s.bAutoUploadSubtitles && m_pCAP && m_pCAP->GetSubtitleDelay() == 0) {
+        m_pSubtitlesProviders->Upload();

```

Comment:
We could check once at the end of the video
the ratio of frames drawn over the total frames of the video > 80%-90%
instead of checking during playback...